src/ui/buttons/buttonDesigns/divideLogoAnimation.jsx

```diff
@@ -0,0 +1,129 @@
+
+    // return null if animation should not be played for the button size
+    const designContainerWidth = designContainer.offsetWidth;
+    if (designContainerWidth < configuration.min || designContainerWidth > configuration.max) {
+        return null;
+    }
+
+    // get the label container that animation will be applied to
+    const paypalLabelContainerElement = designContainer.querySelector(`.${ PAYPAL_BUTTON_LABEL }`) || null;
+    if (!paypalLabelContainerElement) {
+        return null;
+    }
+
+    // get starting position for element so it doesn't flicker when animation begins
+    const paypalLogoElement = (paypalLabelContainerElement && paypalLabelContainerElement.querySelector(`.${ PAYPAL_LOGO }`)) || null;
+    const paypalLogoStartingLeftPosition = paypalLogoElement
+        ? `${ (paypalLogoElement.offsetLeft / paypalLabelContainerElement.offsetWidth) * 100 }%`
+        : '44.5%';
+
+    return {
+        paypalLabelContainerElement,
+        paypalLogoStartingLeftPosition
+    };
+};
+
+function getDivideLogoAnimation(designProps, cssUtilClasses) : void {
+    const { DOM_READY, PAYPAL_LOGO } = cssUtilClasses;
+    const { paypalLabelContainerElement, paypalLogoStartingLeftPosition } = designProps;
+    const designCss = `
+        .${ DOM_READY } .personalized-design-container img.${ PAYPAL_LOGO }{
+            animation: 3s divide-logo-animation-left-side 2s infinite alternate;
+        }
+        
+        .personalized-design-container .personalized-label-container {
+            animation: 3s divide-logo-animation-right-side 2s infinite alternate;
+        }
+
+        @keyframes divide-logo-animation-left-side {
+            0% {
+                position: fixed;
+                left: ${ paypalLogoStartingLeftPosition };
+            }
+            33% {
+                position: fixed;
+                left: ${ paypalLogoStartingLeftPosition };
+            }
+            66% {
+                position: fixed;
+                left: 0%;
+            }
+            100% {
+                position: fixed;
+                left: 0%;
+            }
+        }
+        
+        @keyframes divide-logo-animation-right-side {
+            0%{
+                opacity: 0;
+                position: fixed;
+                right: 45%;
+            }
+            33%{
+                opacity: 0;
+                position: fixed;
+                right: 45%;
+            }
+            66% {
+                opacity: 1;
+                position: fixed;
+                right: 0%;
+            }
+            100% {
+                opacity: 1;
+                position: fixed;
+                right: 0%;
+            }
+        }
+    `;
+
+    if (paypalLabelContainerElement) {
+        const style = document.createElement('style');
+        paypalLabelContainerElement.appendChild(style);
+        style.appendChild(document.createTextNode(designCss));
+    }
+}
+
+export function getDivideLogoAnimationScript() : string {
```

Comment:
Not sure if this can/should be generalized. I'll think about it. The limiting factors would be the specifics of the css, which element the design is applied to, and the fact that it is being passed in as a string

Reply:
I am afraid this script is different for every animation, so creating a util for this is not very useful ( in my humble opinion )